Cookbook

NikitaLavrentiev opened the pull request.

NikitaLavrentiev committed:
дз выполнено

NikitaLavrentiev committed:
Create README.md

NikitaLavrentiev committed:
Update README.md

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/Controllers/FirstController.java:
```diff
@@ -0,0 +1,22 @@
+    @GetMapping("/info")
+    public String info() {
+        return """
+                Ученик: Никита
```
Для переноса строк в конце добавь тег <br>, в html он отвечает за перенос

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/Controllers/FirstController.java:
```diff
@@ -0,0 +1,22 @@
+
+@RestController
+public class FirstController {
+    @GetMapping
```
этот эндпоинт проверял? он в спринге 3 не будет работать, надо вот так написать, чтобы работал: `@GetMapping("/")`

Commit:
Создан класс Recipe (рецепт);
Создан класс Ingredient (ингредиент);
Создан интерфейсы и классы сервисов с методами;
Созданы контроллеры для создания и получения рецептов и ингредиентов.

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/service/IngredientServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+@Service
+public class IngredientServiceImpl implements IngredientService {
+    private static long idCounter = 1;
+    private Map<Long, Ingredient> ingredientMap = new HashMap<>();
```
final

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/service/IngredientServiceImpl.java:
```diff
@@ -0,0 +1,31 @@
+    @Override
+    public Optional<Ingredient> getById(Long id) {
+        return Optional.ofNullable(ingredientMap.get(id));
+    } //Optional защищает от nullPointException
```
если уметь с ним работать)

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/service/RecipeServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+
+public class RecipeServiceImpl implements RecipeService{
+    private static long idCounter = 1;
+    private Map<Long, Recipe> recipeMap = new HashMap<>();
```
final

dmitry-bizin commented on src/main/java/me/lavrentiev/cookbook/service/RecipeServiceImpl.java:
```diff
@@ -0,0 +1,30 @@
+public class RecipeServiceImpl implements RecipeService{
+    private static long idCounter = 1;
+    private Map<Long, Recipe> recipeMap = new HashMap<>();
+    private ValidationService validationService;
```
final
